### PR TITLE
Replaced keys with Profile

### PR DESCRIPTION
**Pull request by SuperMohit**
*Description of changes:*
1. Replaced keys with Profile.
2. Ids are now composite so replaced getAtt with ProjectId.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: templates/mongodb-atlas.base.template.yaml [hunk 1]
```diff
@@ -193,31 +174,33 @@ Resources:
...
             AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
           Action: 'sts:AssumeRole'
+        - Effect: Allow
+          Principal:
+            AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            Service: ecs-tasks.amazonaws.com
+          Action: 'sts:AssumeRole'
+      ManagedPolicyArns:
+        - !Sub arn:${AWS::Partition}:iam::aws:policy/service-role/AmazonECSTaskExecutionRolePolicy
+        - !Sub arn:${AWS::Partition}:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly
```
Comment (andreaangiolillo) [inline templates/mongodb-atlas.base.template.yaml:184]: Why do you need this change?

Comment (SuperMohit) [reply templates/mongodb-atlas.base.template.yaml:184]: @andreaangiolillo see the comment here:
https://github.com/aws-quickstart/quickstart-mongodb-atlas-mean-stack-aws-fargate-integration/pull/1/files#r1112151978

**Review by Zuhairahmed — DISMISSED** [review]

File: .taskcat.yml [hunk 2]
```diff
@@ -17,6 +17,7 @@ tests:
       QSS3BucketRegion: $[taskcat_current_region]
       ClusterName: tcat-$[taskcat_random-string]-$[taskcat_current_region]
       ClusterRegion: US_EAST_1
+      DatabasePassword: override
```
Comment (vsnyc) [inline .taskcat.yml:20]: Can we change this to something like below? Do you have any restrictions on the password values?
```suggestion
      DatabasePassword: $[taskcat_genpass_12]
```

File: .taskcat.yml [hunk 3]
```diff
@@ -32,6 +33,7 @@ tests:
       AvailabilityZones: $[taskcat_genaz_2]
       ClusterName: tcat-$[taskcat_random-string]-$[taskcat_current_region]
       ClusterRegion: US_WEST_1
+      DatabasePassword: override
```
Comment (vsnyc) [inline .taskcat.yml:36]: Can we change this to something like below? Do you have any restrictions on the password values?
```suggestion
      DatabasePassword: $[taskcat_genpass_12]
```

**Review by vsnyc — CHANGES_REQUESTED** [review]
The PR is looking good. Just one minor change requested in .taskcat.yml to change the password value form `override` to `$[taskcat_genpass_12]`. Otherwise, anyone running the code from repo directly will have the password value as `override` if they didn't override it in .taskcat_overrides.yml :)

Also, we need to document the expected value for the Profile parameter: 
* is this the secret name or the ARN? 
* is the pattern `cfn/atlas/profile/{Profile}` suggested or mandatory?
* what is the value of `{Profile}`?
* what is the value contained in the secret?

File: templates/mongodb-atlas-main.template.yaml [hunk 4]
```diff
@@ -26,17 +25,17 @@ Metadata:
...
       - QSS3BucketName
       - QSS3KeyPrefix
       - QSS3BucketRegion
     ParameterLabels:
-      PublicKey:
-        default: MongoDB Atlas API public key
-      PrivateKey:
-        default: MongoDB Atlas API private key
+      Profile:
+        default: "A secret with name cfn/atlas/profile/{Profile}"
```
[the remark this answers is not in the document]
Comment (SuperMohit) [reply templates/mongodb-atlas-main.template.yaml:38]: Yes, the pattern is mandatory. Here is a sample CFN to create that secret.
https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/examples/profile-secret.yaml

**Comment by SuperMohit** [comment]
Latest version of Network Peering has a required field ContainerId, it seems this needs more work. Putting it back to draft.

**Comment by SuperMohit** [comment]
> 

[taskcat_outputs.zip](https://github.com/aws-quickstart/quickstart-mongodb-atlas/files/10882110/taskcat_outputs.zip)
@andreaangiolillo PFA taskcat. For some reason html didn't generate, but it has log file for all 3 regions.

**Comment by SuperMohit** [comment]
Screenshot 2023-03-03 at 7 00 15 PM

**Comment by vsnyc** [comment]
@SuperMohit - thanks for addressing the PR comments, this looks good now. We also need a new PR to the `doc-edits` branch updating the [predeployment steps](https://github.com/aws-quickstart/quickstart-mongodb-atlas/blob/doc-edits/docs/deployment_guide/partner_editable/pre_deployment.adoc) file to indicate that creating the secret first of the pattern: `cfn/atlas/profile/${ProfileName}` is a pre-requisite to installing the Partner Solution. You can link the [sample](https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/examples/profile-secret.yaml) in the documentation.

**Review by vsnyc — DISMISSED** [review]

File: templates/mongodb-atlas-peering.template.yaml [hunk 5]
```diff
@@ -226,19 +220,25 @@ Resources:
...
         - Name: !Sub "${AWS::StackName}"
           Type: "CLUSTER"
+  NetworkContainer:
+    Type: MongoDB::Atlas::NetworkContainer
+    DependsOn: AtlasCluster
+    Properties:
+      Profile:  !Ref "Profile"
+      AtlasCidrBlock: !Ref AtlasCidrBlock
+      RegionName: !Ref ClusterRegion
+      ProjectId: !GetAtt "AtlasProject.Id"
```
Comment (andreaangiolillo) [inline templates/mongodb-atlas-peering.template.yaml:241]: Don't we need also the [network-peering](https://github.com/mongodb/mongodbatlas-cloudformation-resources/tree/master/cfn-resources/network-peering) resource for this template?

Comment (andreaangiolillo) [reply templates/mongodb-atlas-peering.template.yaml:241]: It seems to me that you are only creating a container without setting up the network peering

Comment (SuperMohit) [reply templates/mongodb-atlas-peering.template.yaml:241]: L177

Comment (andreaangiolillo) [reply templates/mongodb-atlas-peering.template.yaml:241]: I missed it. Thanks!

**Comment by poornatejakonathampi** [comment]
> @SuperMohit - thanks for addressing the PR comments, this looks good now. We also need a new PR to the `doc-edits` branch updating the [predeployment steps](https://github.com/aws-quickstart/quickstart-mongodb-atlas/blob/doc-edits/docs/deployment_guide/partner_editable/pre_deployment.adoc) file to indicate that creating the secret first of the pattern: `cfn/atlas/profile/${ProfileName}` is a pre-requisite to installing the Partner Solution. You can link the [sample](https://github.com/mongodb/mongodbatlas-cloudformation-resources/blob/master/examples/profile-secret.yaml) in the documentation.

Created a new PR `https://github.com/aws-quickstart/quickstart-mongodb-atlas/pull/40` for the doc-edits. Updated profile info.

**Review by Zuhairahmed — APPROVED** [review]

**Review by vsnyc — APPROVED** [review]

**Pull request merged**